build: OSX fails to load NIF as symbol _LZ4_compressBound is not found.

I'm guessing this is some sort of missing library reference in the build tools, but I am not clear what to look for or fix specifically, sorry.

```
Symbol not found: _LZ4_compressBound\n  Referenced from: /Volumes/ramdisk/erocksdb/priv/erocksdb.s
```

full log:

```
Last login: Tue Jul 21 12:17:37 on ttys000
Darwin: akai

Agent pid 41899
dch 🍺  /projects ❯❯❯j erocks
/repos/erocksdb
dch 🍺  /r/erocksdb git:develop ❯❯❯make test
./rebar compile
==> erocksdb (compile)
/Library/Developer/CommandLineTools/usr/bin/make  all-am
make[2]: Nothing to be done for `all-am'.
test -z "/repos/erocksdb/c_src/system/lib" || ./install-sh -c -d "/repos/erocksdb/c_src/system/lib"
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libsnappy.la '/repos/erocksdb/c_src/system/lib'
libtool: install: /usr/bin/install -c .libs/libsnappy.1.dylib /repos/erocksdb/c_src/system/lib/libsnappy.1.dylib
libtool: install: (cd /repos/erocksdb/c_src/system/lib && { ln -s -f libsnappy.1.dylib libsnappy.dylib || { rm -f libsnappy.dylib && ln -s libsnappy.1.dylib libsnappy.dylib; }; })
libtool: install: /usr/bin/install -c .libs/libsnappy.lai /repos/erocksdb/c_src/system/lib/libsnappy.la
libtool: install: /usr/bin/install -c .libs/libsnappy.a /repos/erocksdb/c_src/system/lib/libsnappy.a
libtool: install: chmod 644 /repos/erocksdb/c_src/system/lib/libsnappy.a
libtool: install: ranlib /repos/erocksdb/c_src/system/lib/libsnappy.a
test -z "/repos/erocksdb/c_src/system/share/doc/snappy" || ./install-sh -c -d "/repos/erocksdb/c_src/system/share/doc/snappy"
 /usr/bin/install -c -m 644 ChangeLog COPYING INSTALL NEWS README format_description.txt framing_format.txt '/repos/erocksdb/c_src/system/share/doc/snappy'
test -z "/repos/erocksdb/c_src/system/include" || ./install-sh -c -d "/repos/erocksdb/c_src/system/include"
 /usr/bin/install -c -m 644 snappy.h snappy-sinksource.h snappy-stubs-public.h snappy-c.h '/repos/erocksdb/c_src/system/include'
Cloning into 'rocksdb'...
Note: checking out 'rocksdb-3.11.2'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at add2ac5... Bump the version to 3.11.2
  GEN      util/build_version.cc
  GEN      util/build_version.cc
  CC       db/builder.o
  CC       db/c.o
  CC       db/column_family.o
  CC       db/compaction.o
  CC       db/compaction_job.o
  CC       db/compaction_picker.o
  CC       db/db_filesnapshot.o
  CC       db/dbformat.o
  CC       db/db_impl.o
  CC       db/db_impl_debug.o
  CC       db/db_impl_readonly.o
  CC       db/db_impl_experimental.o
  CC       db/db_iter.o
  CC       db/experimental.o
  CC       db/event_logger_helpers.o
  CC       db/file_indexer.o
  CC       db/filename.o
  CC       db/flush_job.o
  CC       db/flush_scheduler.o
  CC       db/forward_iterator.o
  CC       db/internal_stats.o
  CC       db/log_reader.o
  CC       db/log_writer.o
  CC       db/managed_iterator.o
  CC       db/memtable_allocator.o
  CC       db/memtable.o
  CC       db/memtable_list.o
  CC       db/merge_helper.o
  CC       db/merge_operator.o
  CC       db/repair.o
  CC       db/slice.o
  CC       db/table_cache.o
  CC       db/table_properties_collector.o
  CC       db/transaction_log_impl.o
  CC       db/version_builder.o
  CC       db/version_edit.o
  CC       db/version_set.o
  CC       db/wal_manager.o
  CC       db/write_batch.o
  CC       db/write_batch_base.o
  CC       db/write_controller.o
  CC       db/write_thread.o
  CC       port/stack_trace.o
  CC       port/port_posix.o
  CC       table/adaptive_table_factory.o
  CC       table/block_based_filter_block.o
  CC       table/block_based_table_builder.o
  CC       table/block_based_table_factory.o
  CC       table/block_based_table_reader.o
  CC       table/block_builder.o
  CC       table/block.o
  CC       table/block_hash_index.o
  CC       table/block_prefix_index.o
  CC       table/bloom_block.o
  CC       table/cuckoo_table_builder.o
  CC       table/cuckoo_table_factory.o
  CC       table/cuckoo_table_reader.o
  CC       table/flush_block_policy.o
  CC       table/format.o
  CC       table/full_filter_block.o
  CC       table/get_context.o
  CC       table/iterator.o
  CC       table/merger.o
  CC       table/meta_blocks.o
  CC       table/plain_table_builder.o
  CC       table/plain_table_factory.o
  CC       table/plain_table_index.o
  CC       table/plain_table_key_coding.o
  CC       table/plain_table_reader.o
  CC       table/table_properties.o
  CC       table/two_level_iterator.o
  CC       util/arena.o
  CC       util/auto_roll_logger.o
  CC       util/bloom.o
  CC       util/build_version.o
  CC       util/cache.o
  CC       util/coding.o
  CC       util/comparator.o
  CC       util/crc32c.o
  CC       util/db_info_dumper.o
  CC       util/dynamic_bloom.o
  CC       util/env.o
  CC       util/env_hdfs.o
  CC       util/env_posix.o
  CC       util/file_util.o
  CC       util/filter_policy.o
  CC       util/hash.o
  CC       util/hash_cuckoo_rep.o
  CC       util/hash_linklist_rep.o
  CC       util/hash_skiplist_rep.o
  CC       util/histogram.o
  CC       util/instrumented_mutex.o
  CC       util/iostats_context.o
  CC       utilities/backupable/backupable_db.o
  CC       utilities/convenience/convenience.o
  CC       utilities/checkpoint/checkpoint.o
  CC       utilities/compacted_db/compacted_db_impl.o
  CC       utilities/document/document_db.o
  CC       utilities/document/json_document_builder.o
  CC       utilities/document/json_document.o
  CC       utilities/flashcache/flashcache.o
  CC       utilities/geodb/geodb_impl.o
  CC       utilities/leveldb_options/leveldb_options.o
  CC       utilities/merge_operators/put.o
  CC       utilities/merge_operators/string_append/stringappend2.o
  CC       utilities/merge_operators/string_append/stringappend.o
  CC       utilities/merge_operators/uint64add.o
  CC       utilities/redis/redis_lists.o
  CC       utilities/spatialdb/spatial_db.o
  CC       utilities/ttl/db_ttl_impl.o
  CC       utilities/write_batch_with_index/write_batch_with_index.o
  CC       utilities/write_batch_with_index/write_batch_with_index_internal.o
  CC       util/event_logger.o
  CC       util/ldb_cmd.o
  CC       util/ldb_tool.o
  CC       util/log_buffer.o
  CC       util/logging.o
  CC       util/memenv.o
  CC       util/murmurhash.o
  CC       util/mutable_cf_options.o
  CC       util/options_builder.o
  CC       util/options.o
  CC       util/options_helper.o
  CC       util/perf_context.o
  CC       util/rate_limiter.o
  CC       util/skiplistrep.o
  CC       util/slice.o
  CC       util/sst_dump_tool.o
  CC       util/statistics.o
  CC       util/status.o
  CC       util/string_util.o
  CC       util/sync_point.o
  CC       util/thread_local.o
  CC       util/thread_status_impl.o
  CC       util/thread_status_updater.o
  CC       util/thread_status_updater_debug.o
  CC       util/thread_status_util.o
  CC       util/thread_status_util_debug.o
  CC       util/vectorrep.o
  CC       util/xfunc.o
  CC       util/xxhash.o
  AR       librocksdb.a
ar: creating archive librocksdb.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: librocksdb.a(sync_point.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: librocksdb.a(xfunc.o) has no symbols
Compiled src/erocksdb_bump.erl
Compiled src/erocksdb.erl
Compiling c_src/erocksdb.cc
Compiling c_src/refobjects.cc
Compiling c_src/threading.cc
Compiling c_src/workitems.cc
./rebar eunit
==> erocksdb (eunit)
Compiled test/basho_bench_driver_eldb.erl
Compiled test/cleanup.erl
test/rand_gen_1.erl:33: Warning: erlang:now/0: Deprecated BIF. See the "Time and Time Correction in Erlang" chapter of the ERTS User's Guide for more information.
test/rand_gen_1.erl:64: Warning: erlang:now/0: Deprecated BIF. See the "Time and Time Correction in Erlang" chapter of the ERTS User's Guide for more information.
Compiled test/rand_gen_1.erl
Compiled test/basho_bench_driver_rocksdb.erl
Compiled src/erocksdb_bump.erl
Compiled test/iterators.erl
Compiled test/cacheleak.erl
Compiled src/erocksdb.erl
======================== EUnit ========================
undefined
*** test module not found ***
**erocksdb

=======================================================
  Failed: 0.  Skipped: 0.  Passed: 0.
One or more tests were cancelled.

=WARNING REPORT==== 21-Jul-2015::12:27:39 ===
The on_load function for module erocksdb returned {error,
                                                   {load_failed,
                                                    "Failed to load NIF library: 'dlopen(/repos/erocksdb/priv/erocksdb.so, 2): Symbol not found: _LZ4_compressBound\n  Referenced from: /repos/erocksdb/priv/erocksdb.so\n  Expected in: flat namespace\n in /repos/erocksdb/priv/erocksdb.so'"}}
ERROR: One or more eunit tests failed.
make: *** [test] Error 1
dch 🍺  /r/erocksdb git:develop ❯❯❯                                                  ⏎
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-BUILD_CONFIG = build_config.mk
+BUILD_CONFIG = make_config.mk
 
 all: compile
 

